### ISSUE TITLE
Using JS string interpolation inside @raw in template content throws "Unexpected end of template expression"

### ISSUE THREAD

**Issue by dwragge** [casid/jte#166]
Hey,

I was trying to `@raw` to include a script in my page, which was part of a template (a basic root template with `<html>` tags, etc). Inside the script block I used JS string interpolation e.g. ``console.log(`hello ${name}`)``. Unfortunately this threw an exception, detailed below.

Minimal reproducable example:

`testroot.jte`:
```
@import gg.jte.Content

@param Content content

<head>
    <title>Test</title>
</head>
<body>
<div class="content">
    ${content}
</div>
</body>
```
`testraw.jte`:
```
@template.testroot(
content = @`
    <p>Hello World!</p>
    @raw
    <script>
        const hello = "Hello!"
        console.log(`${hello}`)
    </script>
    @endraw
`
)
```

(In the real example, I had included the script as a separate parameter passed to the template, which also failed)

Would you happen to know what's going on here?
Thanks!

Exception encountered:
```
gg.jte.TemplateException: Failed to compile testraw.jte, error at line 7: Unexpected end of template expression
	at gg.jte.compiler.java.JavaCodeGenerator.onError(JavaCodeGenerator.java:221)
	at gg.jte.compiler.TemplateParser.extractTextPart(TemplateParser.java:446)
	at gg.jte.compiler.TemplateParser.doParse(TemplateParser.java:359)
	at gg.jte.compiler.TemplateParser.parse(TemplateParser.java:94)
	at gg.jte.compiler.java.JavaCodeGenerator$ContentProcessor.writeJavaCode(JavaCodeGenerator.java:573)
	at gg.jte.compiler.java.JavaCodeGenerator$ContentProcessor.process(JavaCodeGenerator.java:547)
	at gg.jte.compiler.java.JavaCodeGenerator.writeJavaCodeWithContentSupport(JavaCodeGenerator.java:429)
	at gg.jte.compiler.java.JavaCodeGenerator.appendParam(JavaCodeGenerator.java:482)
	at gg.jte.compiler.java.JavaCodeGenerator.appendParams(JavaCodeGenerator.java:470)
	at gg.jte.compiler.java.JavaCodeGenerator.onTemplateCall(JavaCodeGenerator.java:388)
	at gg.jte.compiler.TemplateParser.lambda$doParse$3(TemplateParser.java:257)
	at gg.jte.compiler.TemplateParser.extract(TemplateParser.java:947)
	at gg.jte.compiler.TemplateParser.doParse(TemplateParser.java:257)
	at gg.jte.compiler.TemplateParser.parse(TemplateParser.java:94)
	at gg.jte.compiler.TemplateParser.parse(TemplateParser.java:89)
	at gg.jte.compiler.TemplateCompiler.generateTemplateCall(TemplateCompiler.java:299)
	at gg.jte.compiler.TemplateCompiler.generate(TemplateCompiler.java:218)
	at gg.jte.compiler.TemplateCompiler.precompile(TemplateCompiler.java:142)
	at gg.jte.compiler.TemplateCompiler.load(TemplateCompiler.java:45)
	...
```

**Comment by casid** [comment]
I just pushed a fix for the `@raw` example you provided. In case you would like to give it a try, you could clone this repository and run a maven clean install on the main branch. Then temporarily set `2.1.2-SNAPSHOT` as jte version in your project.

I'm afraid, the other example you were talking about would still fail. Without `@raw`, the first backtick of the js interpolated string will close the jte content block. Also, `${}` will be interpreted as jte keywords and the Java compiler will complain about the missing variable.

Would it be possible for you to have the JS in a separate script file and reference that from jte? This should also avoid all sorts of security problems when mixing jte with JS. This is the policy we're following at work in jte templates.

**Comment by dwragge** [comment]
Hey @casid, thanks for the prompt fix! I have tested it out locally and the fix works a charm.

I might have explained poorly - the other example I was still using `@raw`, just passing it as another param in the template instead of content. E.g.
```
@param gg.jte.Content title
@param gg.jte.Content content
@param gg.jte.Content script = @``

<!doctype html>
<html lang="en">
<head>
    <title>${title}</title>
    <meta charset="utf-8">
</head>
<body>
<div class="content">
    ${content}
</div>
${script}
</body>
</html>
```

Using `@raw` here with your fix works a charm. Having a separate file would definitely work - but the script is still quite small and having it next to the HTML is kinda nice (and I'm lazy :p) 

Thanks for all your work on JTE - it has been an absolute joy to use!

**Comment by casid** [comment]
Thanks for confirming the fix and for the heads up! Glad you enjoy working with jte :-)

**Comment by casid** [comment]
I just released version [2.1.2](https://github.com/casid/jte/releases/tag/2.1.2) to maven central, including this fix :-)
